create-custom-map-style/README.md

```diff
@@ -0,0 +1,990 @@
+	    callback = callback || function () {};
+	    return callback();
+	  };
+
+	  HttpSource.registerProtocols = function (tilelive) {
+	    tilelive.protocols["http:"] = HttpSource;
+	    tilelive.protocols["https:"] = HttpSource;
+	  };
+
+	  HttpSource.registerProtocols(tilelive);
+
+	  return HttpSource;
+	};
+	```
+13. The module makes use of several environment variables to be able to sign calls to the AWS API. You can run the following to generate the required AWS credentials:
+	```
+	aws sts get-session-token --duration-seconds 36000 --profile Amplify-Default
+	```
+	which will output something like the following
+	```
+	{ 
+	   "Credentials": {
+	        "AccessKeyId": "XXXX0XX0X0XXXXXX0XXX",
+	        "SecretAccessKey": "x0XXxxXXXxXxXX0xXxxX0XXxX0X0xxxx0XxxxxXX",
+	        "SessionToken": "XXxXx0XXxX0xxX0XxXXx//////////....=",
+	        "Expiration": "2021-01-01T00:00:00+00:00"
+	    }
+	}
+	```
+	Set these environment variables in your terminal like so:
+	```
+	export AWS_ACCESS_KEY_ID=<Access Key from Previous Step>
+	export AWS_SECRET_ACCESS_KEY=<Secret Key from Previous Step>
+	export AWS_SESSION_TOKEN=<Session Token from Previous Step>
+	```
+	or optionally run the helper script we've included (be sure to have jq installed first, `brew install jq` on Mac, or follow instructions for other platforms here: https://stedolan.github.io/jq/download/):
+	`./setAwsEnvVariables.sh Amplify-Default`
+
+14. This should be enough to get started using `tessera`. You'll need to provide the full file path to your module, so from your terminal run:
+	```
+	tessera -r $(pwd)/tilelive-minimal.js https://maps.geo.us-west-2.amazonaws.com/maps/v0/maps/CreateCustomMapStyle/tiles/\{z\}/\{x\}/\{y\}
+	```
+15. You should see some output in sdtout like `Listening at http://0.0.0.0:8080`. 
+
+16. Now go back to Maputnik in your browser and click "Data Sources" in the top navigation bar. There should not be any active data sources. If you see any be sure to delete them.
+
+	![Data Sources](media/MaputnikDataSources.png)
+	*Data Sources*
+
+17. We're going to add a new data source as shown below. 		
+	- `Source ID`: `amazon`
+	- `Source Type`: `Vector (XYZ URLs)`
+	- `1st Tile URL`: `http://localhost:8080/{z}/{x}/{y}`
+	- `Min Zoom`: `0`
+	- `Max Zoom`: `22`
+
+	![New Data Source](media/MaputnikNewSource.png)
+	*New Data Source*
+
+18. While we could start from scratch to style the map it is easier to demonstrate with a starter template taken from our map style. Click "Open" in the top navigation bar and under "Upload Style" click "Upload" and select the `src/public/example-style-descriptor.json` file. 
+
+19. This is the style descriptor that is provided by Amazon Location Service as part of your Esri Map. You can download this JSON file by making a signed request to the API endpoint, i.e.
+
+	```
+	https://maps.geo.us-west-2.amazonaws.com/maps/v0/maps/CreateCustomMapStyle/style-descriptor?<YOUR SIGV4 QUERY VARIABLES>
```

Comment:
`aws location get-map-style-descriptor --map-name CreateCustomMapStyle` may be an easier way to get this.